### PR TITLE
Adds async search API request and response examples

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

Related to https://github.com/elastic/elasticsearch-specification/issues/2482.
This PR adds request and response examples to the async search API.

### CODE REVIEW

File: specification/async_search/submit/AsyncSearchSubmitResponseExample1.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "summary": "A successful response when performing search asynchronously.",
+  "description": "",
+  "type": "response",
+  "response_code": 200,
+  "value": "{\n  \"id\" : \"FmRldE8zREVEUzA2ZVpUeGs2ejJFUFEaMkZ5QTVrSTZSaVN3WlNFVmtlWHJsdzoxMDc=\", <1>\n  \"is_partial\" : true, <2>\n  \"is_running\" : true, <3>\n  \"start_time_in_millis\" : 1583945890986,\n  \"expiration_time_in_millis\" : 1584377890986,\n  \"response\" : {\n    \"took\" : 1122,\n    \"timed_out\" : false,\n    \"num_reduce_phases\" : 0,\n    \"_shards\" : {\n      \"total\" : 562, <4>\n      \"successful\" : 3, <5>\n      \"skipped\" : 0,\n      \"failed\" : 0\n    },\n    \"hits\" : {\n      \"total\" : {\n        \"value\" : 157483, <6>\n        \"relation\" : \"gte\"\n      },\n      \"max_score\" : null,\n      \"hits\" : [ ]\n    }\n  }\n}"
```
Comment (lcawl) [inline specification/async_search/submit/AsyncSearchSubmitResponseExample1.json:6]: >   "value": "{\n  \"id\" : \"FmRldE8zREVEUzA2ZVpUeGs2ejJFUFEaMkZ5QTVrSTZSaVN3WlNFVmtlWHJsdzoxMDc=\", <1>\n  \"is_partial\" : true, <2>\n  \"is_running\" : true, <3>\n  \"start_time_in_millis\" : 1583945890986,\n  

Looks like there are some footnotes in this example, which should be removed IMO

**Comment by lcawl** [comment]
I added a commit that appends overlays for the async_search.submit operation. I ran `make overlay-docs` then linted the output and it has the following warnings:

>  62541:27      warning  oas3-schema                 Property "method_request" is not expected to be here.                               components.examples.AsyncSearchSubmitRequestExample1.method_request
 62543:17      warning  oas3-schema                 Property "type" is not expected to be here.                                         components.examples.AsyncSearchSubmitRequestExample1.type
 62549:17      warning  oas3-schema                 Property "type" is not expected to be here.                                         components.examples.AsyncSearchSubmitResponseExample1.type
 62550:26      warning  oas3-schema                 Property "response_code" is not expected to be here.                                components.examples.AsyncSearchSubmitResponseExample1.response_code

However when I preview the output in Bump.sh, the example showed up fine.

File: specification/async_search/get/AsyncSearchGetResponseExample1.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "summary": "A succesful response when retrieving the results of a previously submitted async search request given its identifier.",
+  "description": "",
+  "type": "response",
+  "response_code": 200,
+  "value": "{\n  \"id\" : \"FmRldE8zREVEUzA2ZVpUeGs2ejJFUFEaMkZ5QTVrSTZSaVN3WlNFVmtlWHJsdzoxMDc=\",\n  \"is_partial\" : false, <1>\n  \"is_running\" : false, <2>\n  \"start_time_in_millis\" : 1583945890986,\n  \"expiration_time_in_millis\" : 1584377890986, <3>\n  \"completion_time_in_millis\" : 1583945903130, <4>\n  \"response\" : {\n    \"took\" : 12144,\n    \"timed_out\" : false,\n    \"num_reduce_phases\" : 46, <5>\n    \"_shards\" : {\n      \"total\" : 562,\n      \"successful\" : 188, <6>\n      \"skipped\" : 0,\n      \"failed\" : 0\n    },\n    \"hits\" : {\n      \"total\" : {\n        \"value\" : 456433,\n        \"relation\" : \"eq\"\n      },\n      \"max_score\" : null,\n      \"hits\" : [ ]\n    },\n    \"aggregations\" : { <7>\n      \"sale_date\" :  {\n        \"buckets\" : []\n      }\n    }\n  }\n}"
+}
```
Comment (szabosteve) [inline specification/async_search/get/AsyncSearchGetResponseExample1.json:7]:
```suggestion
  "value": "{\n  \"id\" : \"FmRldE8zREVEUzA2ZVpUeGs2ejJFUFEaMkZ5QTVrSTZSaVN3WlNFVmtlWHJsdzoxMDc=\",\n  \"is_partial\" : false,\n  \"is_running\" : false,\n  \"start_time_in_millis\" : 1583945890986,\n  \"expiration_time_in_millis\" : 1584377890986,\n  \"completion_time_in_millis\" : 1583945903130,\n  \"response\" : {\n    \"took\" : 12144,\n    \"timed_out\" : false,\n    \"num_reduce_phases\" : 46,\n    \"_shards\" : {\n      \"total\" : 562,\n      \"successful\" : 188,\n      \"skipped\" : 0,\n      \"failed\" : 0\n    },\n    \"hits\" : {\n      \"total\" : {\n        \"value\" : 456433,\n        \"relation\" : \"eq\"\n      },\n      \"max_score\" : null,\n      \"hits\" : [ ]\n    },\n    \"aggregations\" : {\n      \"sale_date\" :  {\n        \"buckets\" : []\n      }\n    }\n  }\n}"
}
```

File: specification/async_search/get/AsyncSearchGetResponseExample2.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "summary": "A succesful response when retrieving the status of a previously submitted and completed async search without the results.",
+  "description": "For an async search that has been completed, the status response has an additional `completion_status` field that shows the status code of the completed async search which was – in this case – a success.",
+  "type": "response",
+  "response_code": 200,
+  "value": "{\n  \"id\" : \"FmRldE8zREVEUzA2ZVpUeGs2ejJFUFEaMkZ5QTVrSTZSaVN3WlNFVmtlWHJsdzoxMDc=\",\n  \"is_running\" : false,\n  \"is_partial\" : false,\n  \"start_time_in_millis\" : 1583945890986,\n  \"expiration_time_in_millis\" : 1584377890986,\n  \"_shards\" : {\n      \"total\" : 562,\n      \"successful\" : 562,\n      \"skipped\" : 0,\n      \"failed\" : 0\n  },\n \"completion_status\" : 200 <1>\n}"
```
Comment (szabosteve) [inline specification/async_search/get/AsyncSearchGetResponseExample2.json:6]:
```suggestion
  "value": "{\n  \"id\" : \"FmRldE8zREVEUzA2ZVpUeGs2ejJFUFEaMkZ5QTVrSTZSaVN3WlNFVmtlWHJsdzoxMDc=\",\n  \"is_running\" : false,\n  \"is_partial\" : false,\n  \"start_time_in_millis\" : 1583945890986,\n  \"expiration_time_in_millis\" : 1584377890986,\n  \"_shards\" : {\n      \"total\" : 562,\n      \"successful\" : 562,\n      \"skipped\" : 0,\n      \"failed\" : 0\n  },\n \"completion_status\" : 200\n}"
```

File: specification/async_search/status/AsyncSearchStatusResponseExample1.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "summary": "A succesful response when retrieving the status of a previously submitted async search without the results.",
+  "description": "",
+  "type": "response",
+  "response_code": 200,
+  "value": "{\n  \"id\" : \"FmRldE8zREVEUzA2ZVpUeGs2ejJFUFEaMkZ5QTVrSTZSaVN3WlNFVmtlWHJsdzoxMDc=\",\n  \"is_running\" : true,\n  \"is_partial\" : true,\n  \"start_time_in_millis\" : 1583945890986,\n  \"expiration_time_in_millis\" : 1584377890986,\n  \"_shards\" : {\n      \"total\" : 562,\n      \"successful\" : 188, <1>\n      \"skipped\" : 0,\n      \"failed\" : 0\n  }\n}"
+}
```
Comment (szabosteve) [inline specification/async_search/status/AsyncSearchStatusResponseExample1.json:7]:
```suggestion
  "value": "{\n  \"id\" : \"FmRldE8zREVEUzA2ZVpUeGs2ejJFUFEaMkZ5QTVrSTZSaVN3WlNFVmtlWHJsdzoxMDc=\",\n  \"is_running\" : true,\n  \"is_partial\" : true,\n  \"start_time_in_millis\" : 1583945890986,\n  \"expiration_time_in_millis\" : 1584377890986,\n  \"_shards\" : {\n      \"total\" : 562,\n      \"successful\" : 188,\n      \"skipped\" : 0,\n      \"failed\" : 0\n  }\n}"
}
```

File: specification/async_search/submit/AsyncSearchSubmitResponseExample1.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+}
```
Comment (szabosteve) [inline specification/async_search/submit/AsyncSearchSubmitResponseExample1.json:7]:
```suggestion
  "value": "{\n  \"id\" : \"FmRldE8zREVEUzA2ZVpUeGs2ejJFUFEaMkZ5QTVrSTZSaVN3WlNFVmtlWHJsdzoxMDc=\",\n  \"is_partial\" : true,\n  \"is_running\" : true,\n  \"start_time_in_millis\" : 1583945890986,\n  \"expiration_time_in_millis\" : 1584377890986,\n  \"response\" : {\n    \"took\" : 1122,\n    \"timed_out\" : false,\n    \"num_reduce_phases\" : 0,\n    \"_shards\" : {\n      \"total\" : 562,\n      \"successful\" : 3,\n      \"skipped\" : 0,\n      \"failed\" : 0\n    },\n    \"hits\" : {\n      \"total\" : {\n        \"value\" : 157483,\n        \"relation\" : \"gte\"\n      },\n      \"max_score\" : null,\n      \"hits\" : [ ]\n    }\n  }\n}"
}
```

**Review by lcawl — APPROVED** [review]
lgtm

**Pull request merged**